webapp/components/suggestion/at_mention_provider.jsx

```diff
@@ -98,60 +99,89 @@ class AtMentionSuggestion extends Suggestion {
+            break;
+        }
 
-            const filtered = [];
+        const user = users[id];
 
-            for (const id of Object.keys(users)) {
-                const user = users[id];
+        if (user.delete_at > 0) {
+            continue;
+        }
 
-                if (user.delete_at > 0) {
-                    continue;
-                }
+        if (user.username.startsWith(prefix) ||
+            (user.first_name && user.first_name.toLowerCase().startsWith(prefix)) ||
+            (user.last_name && user.last_name.toLowerCase().startsWith(prefix)) ||
+            (user.nickname && user.nickname.toLowerCase().startsWith(prefix))) {
+            filtered.push(user);
+        }
+    }
 
-                if (user.username.startsWith(prefix) ||
-                    (user.first_name && user.first_name.toLowerCase().startsWith(prefix)) ||
-                    (user.last_name && user.last_name.toLowerCase().startsWith(prefix)) ||
-                    (user.nickname && user.nickname.toLowerCase().startsWith(prefix))) {
-                    filtered.push(user);
-                }
+    return filtered;
+}
 
-                if (filtered.length >= MaxUserSuggestions) {
-                    break;
-                }
-            }
+export default class AtMentionProvider {
+    handlePretextChanged(suggestionId, pretext) {
+        const captured = (/@([a-z0-9\-\._]*)$/i).exec(pretext.toLowerCase());
+        if (captured) {
+            const prefix = captured[1];
 
-            if (!pretext.startsWith('/msg')) {
-                // add dummy users to represent the @channel and @all special mentions when not using the /msg command
-                if ('channel'.startsWith(prefix)) {
-                    filtered.push({username: 'channel'});
-                }
-                if ('all'.startsWith(prefix)) {
-                    filtered.push({username: 'all'});
-                }
-                if ('here'.startsWith(prefix)) {
-                    filtered.push({username: 'here'});
+            // Group users into members and nonmembers of the channel.
+            const users = UserStore.getActiveOnlyProfiles(true);
+            const channelMembers = {};
+            const extra = ChannelStore.getCurrentExtraInfo();
+            for (let i = 0; i < extra.members.length; i++) {
+                const id = extra.members[i].id;
+                if (users[id]) {
+                    channelMembers[id] = users[id];
+                    Reflect.deleteProperty(users, id);
                 }
             }
+            const channelNonmembers = users;
+
+            // Filter users by prefix.
+            const filteredMembers = filterUsersByPrefix(
+                    channelMembers, prefix, MaxUserSuggestions);
+            const filteredNonmembers = filterUsersByPrefix(
+                    channelNonmembers, prefix, MaxUserSuggestions - filteredMembers.length);
+            const filteredSpecialMentions = pretext.startsWith('/msg') ?
```

Comment:
Do you mind switching this to use a regular if statement instead of a ternary? We're trying to avoid ternary except in really simple, one-line cases. Otherwise, it looks good.